wa/workloads/motionmark/__init__.py

```diff
@@ -0,0 +1,81 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import re
+
+from wa import UiautoWorkload, Parameter
+from wa.framework.exception import ValidationError, WorkloadError
+from wa.utils.types import list_of_strs
+from wa.utils.misc import unique
+
+
+class Motionmark(UiautoWorkload):
+
+    name = 'motionmark'
+
+    description = '''
+    A workload to execute the motionmark web based benchmark
+
+    MotionMark is a graphics benchmark that measures a browser capability to animate complex scenes at a target frame rate
+
+    Test description:
+    1. Open browser application
+    2. Navigate to the motionmark website - http://browserbench.org/motionmark/
+    3. Execute the benchmark
+    '''
+
+    requires_network = True
+
+    regex = [re.compile(r'Multiply Score (.+)'),
+                    re.compile(r'Canvas Score (.+)'),
+                    re.compile(r'Leaves Score (.+)'),
+                    re.compile(r'Paths Score (.+)'),
+                    re.compile(r'Canvas Lines Score (.+)'),
+                    re.compile(r'Focus Score (.+)'),
+                    re.compile(r'Images Score (.+)'),
+                    re.compile(r'Design Score (.+)'),
+                    re.compile(r'Suits Score (.+)')]
+    score_regex = re.compile(r'.*?([\d.]+).*')
+
+    def __init__(self, target, **kwargs):
+        super(Motionmark, self).__init__(target, **kwargs)
+        self.gui.timeout = 1500
+
+    def setup(self, context):
+        super(Motionmark, self).setup(context)
+        self.target.open_url('https://browserbench.org/MotionMark/')
+
+    def update_output(self, context):
+        super(Motionmark, self).update_output(context)
+        num_unprocessed_results = len(self.regex)
+        logcat_file = context.get_artifact_path('logcat')
+        with open(logcat_file) as fh:
+            for line in fh:
+                for regex in self.regex:
+                    match = regex.search(line)
+                    # Check if we have matched the score string in logcat
+                    if match:
+                        score_match = self.score_regex.search(match.group(1))
+                        # Check if there is valid number found for the score.
+                        if score_match:
+                            result = float(score_match.group(1))
+                        else:
+                            result = float('NaN')
+                        entry = regex.pattern.rsplit(None, 1)[0]
+                        context.add_metric(entry, result, 'Score', lower_is_better=False)
+                        num_unprocessed_results -= 1
+        if num_unprocessed_results > 0:
+            msg = "The Motionmark workload has failed. Expected {} scores, Detected {} scores."
+            raise WorkloadError(msg.format(len(self.regex), num_unprocessed_results))
```

Comment:
I don't think so. 

We set num_unprocessed _results = len(self.regex) on line 62 so at this point it would be 9. 

After a entry is processed correctly and added we minus the num_unprocessed_results by 1. 

If all are processed correctly num_unprocessed_results will be 0. 

The exception is then raised if num_unprocessed_results is > 0.

Reply:
Sorry I agree the logic of raising the error if correct, I meant in the message itself it lists the number of expected scores and the number of "detected scores" which in this case would actually be reporting the inverse, i.e. how many score were not detected. I would suggest either rewording the error message or adjusting the value reported.